lib/manager/mix/extract.js

```diff
@@ -0,0 +1,68 @@
+const { isValid } = require('../../versioning/hex');
+
+module.exports = {
+  extractPackageFile,
+};
+
+function extractPackageFile(content) {
+  logger.trace('mix.extractPackageFile()');
+  const deps = [];
+  const contentArr = content.split('\n');
+  const depSectionRegExp = /defp\s+deps.*do/g;
+  const depMatchRegExp = /{:(\w+),\s*([^:"]+)?:?\s*"([^"]+)",?\s*(organization: "(.*)")?.*}/gm;
+
+  for (let lineNumber = 0; lineNumber < contentArr.length; lineNumber += 1) {
+    if (contentArr[lineNumber].match(depSectionRegExp)) {
+      logger.trace(`Matched dep section on line ${lineNumber}`);
+      let depBuffer = '';
+      do {
+        depBuffer += contentArr[lineNumber] + '\n';
+        lineNumber += 1;
+      } while (!contentArr[lineNumber].includes('end'));
+      let depMatch;
+      do {
+        depMatch = depMatchRegExp.exec(depBuffer);
+        if (depMatch) {
+          const depName = depMatch[1];
+          const datasource = depMatch[2];
+          const currentValue = depMatch[3];
+          const organization = depMatch[5];
+
+          const dep = {
+            depName,
+            currentValue,
+          };
+
+          if (datasource) {
+            dep.datasource = datasource;
```

Comment:
What are the possible values of "datasource" here? We should only set it, if it's a datasource we support.

Reply:
It can be "github"  so it references a github repo

Reply:
In this case check the datasource explicitly, e.g. `if (datasource === 'github')`. We do not want any "random" datasources being returned, it must be a supported one.